kratos/input_output/cad_json_input.h

```diff
@@ -527,11 +527,13 @@ class CadJsonInput : public IO
                     knot_vector,
                     control_point_weights));
         }
-        return Kratos::make_shared<NurbsCurveGeometry<TWorkingSpaceDimension, PointerVector<TThisNodeType>>>(
-            NurbsCurveGeometry<TWorkingSpaceDimension, PointerVector<TThisNodeType>>(
+        typename NurbsCurveGeometry<TWorkingSpaceDimension, PointerVector<TThisNodeType>>::Pointer p_nurbs_curve(

```

Comment:
why does this make a difference?
Can you explain please?

Reply:
Thanks for checking the PR.

Once those geometries are created they assign its own Id dependent on its pointer. And with the other (probably more suggested constructor) this Id would always be the same and lateron the Ids aren't unique anymore.

https://github.com/KratosMultiphysics/Kratos/blob/c46c9420cd32d8557d8d3912c58a3c39fa34300a/kratos/geometries/geometry.h#L3929

It is questionable if this is the best way to self assign Ids, however, with the transformed pointer constructor they always have unique ids...

Reply:
ok I think I understand 

Can you please add a test if possible? Otherwise I fear that this is quite fragile and might break in the future while refactoring

Reply:
true. this might be failing pretty easy. I added tests, which are coming with #8886